mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java

```diff
@@ -147,15 +156,49 @@ public void storeAttachment(Attachment attachment) throws MailboxException {
         );
     }
 
+    private CompletableFuture<Void> asyncStoreAttachmentForMessageId(Attachment attachment, MessageId ownerMessageId) throws IOException {
+        return cassandraAsyncExecutor.executeVoid(
+            update(CassandraAttachmentMessageIdTable.TABLE_NAME)
+                .with(add(CassandraAttachmentMessageIdTable.MESSAGE_IDS, ownerMessageId.serialize()))
+                .where(eq(CassandraAttachmentMessageIdTable.ATTACHMENT_ID, attachment.getAttachmentId().getId())));
+    }
+
     @Override
-    public void storeAttachments(Collection<Attachment> attachments) throws MailboxException {
+    public void storeAttachmentsForMessage(Collection<Attachment> attachments, MessageId ownerMessageId) throws MailboxException {
         try {
             FluentFutureStream.of(
-                attachments.stream()
-                    .map(Throwing.function(this::asyncStoreAttachment)))
+                Stream.concat(
+                    attachments.stream()
+                        .map(Throwing.function(this::asyncStoreAttachment)),
+                    attachments.stream()
```

Comment:
I don't think both actions should be done at the same time. What if asyncStoreAttachment fails ?

Reply:
done, each action is now separated and chained by thenCompose